art/estimators/classification/keras.py

```diff
@@ -499,6 +524,28 @@ def get_activations(self, x, layer, batch_size):
 
         return activations
 
+    def custom_gradient(self, nn_function, tensors, input_values):
+        """
+        Returns the gradient of the nn_function with respect to model input
+
+        :param nn_function: an intermediate tensor representation of the function to differentiate
+        :type nn_function: a Keras tensor
+        :param tensors: the tensors or variables to differentiate with respect to
+        :type tensors: `list`
+        :param input_values: the inputs to evaluate the gradient
+        :type input_values: `list`
+        :return: the gradient of the function w.r.t vars
+        :rtype: `np.ndarray`
+        """
+        import keras.backend as k
+
+        grads = k.gradients(nn_function, tensors[0])[0]
+        outputs = k.function(tensors, [grads])
```

Comment:
This approach lead to a memory leak. Please follow the approach in PR #417

Reply:
This seems difficult to replicate since one of the tensors (namely the `poison_placeholder` tensor declared in the forward step) is always changing, so the function itself couldn't be cached. Not sure how to move forward with that.

Reply:
Fixed in 5144abd